### PR TITLE
Adopt the Strimzi Drain Cleaner project

### PR DESCRIPTION
**Pull request by scholzj**
This proposal suggests to adopt the [Strimzi Drain Cleaner](https://github.com/scholzj/strimzi-cistic-odpadu) project under the Strimzi umbrella. That should help to make it better integrated in out docs and system tests and make users more aware of it.

_Disclaimer: I'm the original author of the project._

### CODE REVIEW

File: 029-adopt-the-drain-cleaner-utility.md
Comment (ppatierno) [inline 029-adopt-the-drain-cleaner-utility.md:20]: "and" to be removed?

File: 029-adopt-the-drain-cleaner-utility.md
Comment (ppatierno) [inline 029-adopt-the-drain-cleaner-utility.md:40]: a link to documentation about it could be useful here.

**Review by samuel-hawker — COMMENTED** [review]
This seems like a great bit of value to users, added some minor grammar suggestions.

I wonder if `drain-cleaner` is the best name for this, or if something like `kafka-safe-disruption-drainer` might be more accurate? More specifically not sure about the 'cleaner' part here, maybe I am missing something though in the implementation or in the context that would make it clean?

**Review by ppatierno — APPROVED** [review]

**Review by sknot-rh — APPROVED** [review]

**Comment by scholzj** [comment]
> I wonder if drain-cleaner is the best name for this, or if something like kafka-safe-disruption-drainer might be more accurate? More specifically not sure about the 'cleaner' part here, maybe I am missing something though in the implementation or in the context that would make it clean?

@samuel-hawker It is called drain cleaner because when you set the `maxUnavailable` in PDB to `0`, it will block the draining of the worker nodes. And the drain cleaner toll will unblock it => clean it. I'm open to other name suggestions, u TBH `kafka-safe-disruption-drainer` is equally cryptic and just much longer. I do not think you can give it a name from which the user will understand what it really does. and why to use it.

**Comment by samuel-hawker** [comment]
You're right that my name isn't better! :P I think leaving as is works- just curious to see if yourself or the other maintainers had an inspirational name alterative, bu I doubt we will find one

**Review by samuel-hawker — APPROVED** [review]

**Comment by scholzj** [comment]
> You're right that my name isn't better! :P I think leaving as is works- just curious to see if yourself or the other maintainers had an inspirational name alterative, bu I doubt we will find one

If you don't like the `drain-cleaner` or think it is particularly confusing, maybe we can use something simpler ... `drainer` or `pod-drainer` 🤷‍♂️

**Comment by samuel-hawker** [comment]
I'll give it some thought. Initial thoughts are something like `kafka-disruption-drainer`
but i think drain-cleaner is serviceable.

**Review by tombentley — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**
